include/imgproc/dct_hash.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+// Standard C++ includes
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// Standard C includes
+#include <cmath>
+
+//opencv
+#include <opencv2/opencv.hpp>
+
+
+class DCTHash {
+
+    private:
+    static unsigned int countSetBits(unsigned long long int n) { 
+        unsigned int count = 0; 
+        while (n) { 
+            count += n & 1; 
+            n >>= 1; 
+        } 
+        return count; 
+    } 
+
+    public:
+    
+    
+    //! computes the DCT hash 
+    static unsigned long long computeHash(const cv::Mat &in) {
+        
+		unsigned long long hash;
+        cv::Mat dct_mat;
+        cv::dct(in, dct_mat);
+        cv::Rect roi( 0, 0, 8, 8); 
+        cv::Mat low_dct( dct_mat, roi ); 
+
+		std::vector<float> vec;
+		for (int row = 0; row < 9; row++) {
+			for (int col = 0; col < 9; col++) {
+			vec.push_back(low_dct.at<float>(row,col));
+			}
+		}
+
+		// calculate median value 
+		std::sort(std::begin(vec), std::end(vec));
+		float medianv;
+		int n = vec.size();
+		if (n % 2 != 0) {
+			medianv = vec[n/2];
+		} else {
+			medianv = (vec[(n-1)/2] + vec[n/2]) / 2;
+		}
+		
+		
+		const cv::Mat mask = (low_dct > medianv);
+		const cv::Mat row = mask.reshape(1,1)/255;
+
+		std::vector<int> hashvector;
+		row.row(0).copyTo(hashvector);
+		std::stringstream ss;
+		std::string binstring;
+		for (int v : hashvector) { ss << v; }
+		ss >> binstring;
+		unsigned long long int binary = std::stoull(binstring, 0, 2);
+		hash = binary;
+		return hash;
```

Comment:
This is a fairly roundabout way of achieving this...

How about:
```c++
unsigned long long hash = 0;
for (size_t i = 0; i < 64; i++) {
    if(reinterpret_cast<int *>(mask.data)[i]) {
        hash |= 1 << i;
    }
}
```
(I'm assuming that ``mask`` is an array of ``int``s.)

Reply:
I've just realised we don't actually need ``mask`` at all here:
```c++
unsigned long long hash = 0;
for (size_t i = 0; i < 64; i++) {
    if(reinterpret_cast<float *>(low_dct.data)[i] > medianv) {
        hash |= 1 << i;
    }
}
```

Reply:
Excellent point

Reply:
good point, the current implementation is just a hack with the strings and stuff